src/vns_lprc.jl

```diff
@@ -0,0 +1,227 @@
+function perturbation_VNS_LPRC(solution::Solution, p::Int, k::Int, instance::Instance)
+    if p == 1
+        return perturbation_VNS_LPRC_exchange(solution, k, instance)
+    else
+        return perturbation_VNS_LPRC_insertion(solution, k, instance)
+    end
+end
+
+# The Local Search is based on a car exchange move.
+function localSearch_VNS_LPRC!(solution::Solution, perturbation_exchange::Bool, instance::Instance)
+
+    # useful variables
+    nb_vehicles = length(solution.sequence)
+    b0 = instance.nb_late_prec_day+1
+
+    improved = true
+    while improved
+        phi = cost_VNS_LPRC(solution, instance)
+        for i in b0:nb_vehicles
+            best_delta = 0
+            list = Array{Int, 1}()
+            for j in (i+1):nb_vehicles # exchange (i, j) is the same as exchange (j, i)
+                if !perturbation_exchange || same_HPRC(solution, i, j, instance)
+                    delta = cost_move_exchange(solution, i, j, instance, 2)
+                    if delta < best_delta
+                        list = [j]
+                        best_delta = delta
+                    elseif delta == best_delta
+                        push!(list, j)
+                    end
+                end
+            end
+            if list != []
+                k = rand(list)
+                move_exchange!(solution, i, k, instance)
+            end
+        end
+        if phi == cost_VNS_LPRC(solution, instance)
+            improved = false
+        end
+    end
+end
+
+
+function localSearch_intensification_VNS_LPRC!(solution::Solution, alpha::Int, cost_move::Function, move!::Function, instance::Instance)
+    # useful variables
+    nb_vehicles = length(solution.sequence)
+    b0 = instance.nb_late_prec_day+1
+
+    nb_non_improved = 0
+    while nb_non_improved < alpha
+        phi = cost_VNS_LPRC(solution, instance)
+        for i in b0:nb_vehicles
+            best_delta = 0
+            list = Array{Int, 1}()
+            for j in b0:nb_vehicles
+                delta = cost_move(solution, i, j, instance, 2)
+                if delta < best_delta
+                    list = [j]
+                    best_delta = delta
+                elseif delta == best_delta
+                    push!(list, j)
+                end
+            end
+            if list != []
+                k = rand(list)
+                move!(solution, i, k, instance)
+            end
+        end
+        nb_non_improved += 1
+        if phi < cost_VNS_LPRC(solution, instance)
+            nb_non_improved = 0
+        end
+    end
+end
+
+# Apply two local search, first one with insertion move, and the second one with exchange move.
+function intensification_VNS_LPRC!(solution::Solution, instance::Instance)
+    localSearch_intensification_VNS_LPRC!(solution, VNS_LPRC_ALPHA_PERTURBATION, cost_move_insertion, move_insertion!, instance)
+    localSearch_intensification_VNS_LPRC!(solution, VNS_LPRC_ALPHA_PERTURBATION, cost_move_exchange, move_exchange!, instance)
```

Comment:
add `return solution`

Reply:
Why should I ? Intensification is supposed to increase the current solution, not to return an other one I think.

Reply:
This is just something we can add. See [this](https://docs.julialang.org/en/v1/manual/style-guide/index.html#Append-!-to-names-of-functions-that-modify-their-arguments-1): _It is typical for such functions to also return the modified array for convenience._